openfe/protocols/restraint_utils/geometry/boresch.py

```diff
@@ -0,0 +1,870 @@
+    guest_restraint_atoms_idxs : Optional[list[int]]
+      User selected indices of the guest molecule itself (i.e. indexed
+      starting a 0 for the guest molecule). This overrides the
+      restraint search and a restraint using these indices will
+      be retruned. Must be defined alongside ``host_restraint_atoms_idxs``.
+    host_restraint_atoms_idxs : Optional[list[int]]
+      User selected indices of the host molecule itself (i.e. indexed
+      starting a 0 for the hosts molecule). This overrides the
+      restraint search and a restraint using these indices will
+      be returnned. Must be defined alongside ``guest_restraint_atoms_idxs``.
+    host_selection : str
+      An MDAnalysis selection string to sub-select the host atoms.
+    dssp_filter : bool
+      Whether or not to filter the host atoms by their secondary structure.
+    rmsf_cutoff : unit.Quantity
+      The cutoff value for atom root mean square fluction. Atoms with RMSF
+      values above this cutoff will be disregarded.
+      Must be in units compatible with nanometer.
+    host_min_distance : unit.Quantity
+      The minimum distance between any host atom and the guest G0 atom.
+      Must be in units compatible with nanometer.
+    host_max_distance : unit.Quantity
+      The maximum distance between any host atom and the guest G0 atom.
+      Must be in units compatible with nanometer.
+    angle_force_constant : unit.Quantity
+      The force constant for the G1-G0-H0 and G0-H0-H1 angles. Must be
+      in units compatible with kilojoule / mole / radians ** 2.
+    temperature : unit.Quantity
+      The system temperature in units compatible with Kelvin.
+
+    Returns
+    -------
+    BoreschRestraintGeometry
+      An object defining the parameters of the Boresch-like restraint.
+
+    References
+    ----------
+    [1] Baumann, Hannah M., et al. "Broadening the scope of binding free energy
+        calculations using a Separated Topologies approach." (2023).
+    """
+    u = mda.Universe(
+        topology,
+        trajectory,
+        format=_get_mda_coord_format(trajectory),
+        topology_format=_get_mda_topology_format(topology),
+    )
+
+    if (guest_restraint_atoms_idxs is not None) and (host_restraint_atoms_idxs is not None):  # fmt: skip
+        # In this case assume the picked atoms were intentional /
+        # representative of the input and go with it
+        guest_ag = u.select_atoms[guest_idxs]
+        guest_anchor = [
+            at.ix for at in guest_ag.atoms[guest_restraint_atoms_idxs]
+        ]
+        host_ag = u.select_atoms[host_idxs]
+        host_anchor = [
+            at.ix for at in host_ag.atoms[host_restraint_atoms_idxs]
+        ]
+
+        # Set the equilibrium values as those of the final frame
+        u.trajectory[-1]
+        atomgroup = u.atoms[host_anchor + guest_anchor]
+        bond, ang1, ang2, dih1, dih2, dih3 = _get_restraint_distances(
+            atomgroup
+        )
+
```

Comment:
TODO: add in checks here so we can warn if it's a bad pick. May push this to a follow-up issue.